### PR TITLE
Only run filters and blocks client-side on remote feeds.

### PR DESCRIPTION
**Pull request by derspyy**
this choice was made because on authenticated requests, these are handled by the server.

### CODE REVIEW

File: Mammoth/Screens/HomeScreen/NewsFeedViewModel+Services.swift [hunk 1]
```diff
@@ -37,8 +37,16 @@ extension NewsFeedViewModel {
 
                 if let lastId = self.oldestItemId(forType: currentType) {
                     let (items, cursorId) = try await currentType.fetchAll(range: RequestRange.max(id: lastId, limit: 20), batchName: "next-page_batch")
-  
-                    let newItems = items.removeMutesAndBlocks().removeFiltered()
+                    
+                    // only remove mutes and blocks in remote feeds.
+                    let newItems: [NewsFeedListItem]
+                    if case .community = type {
```
Comment (bnolens) [inline Mammoth/Screens/HomeScreen/NewsFeedViewModel+Services.swift:43]: Also for `.channel(let type)`, right?

**Review by bnolens — CHANGES_REQUESTED** [review]

File: Mammoth/Screens/HomeScreen/NewsFeedViewModel+Services.swift (hunk 1 above)
Comment (derspyy) [reply Mammoth/Screens/HomeScreen/NewsFeedViewModel+Services.swift:43]: oh! are those... smart lists?

Comment (bnolens) [reply Mammoth/Screens/HomeScreen/NewsFeedViewModel+Services.swift:43]: Yes

Comment (bnolens) [reply Mammoth/Screens/HomeScreen/NewsFeedViewModel+Services.swift:43]: We called them channels before launch

**Review by bnolens — APPROVED** [review]
LGTM!

**Pull request merged**